modules/spring-config/src/test/java/org/mule/runtime/config/internal/LazyComponentInitializerAdapterTestCase.java

```diff
@@ -0,0 +1,240 @@
+
+  private ElementDeclarer declarer;
+
+  @Mock
+  private ExtensionManager extensionManager;
+
+  @Mock
+  private OptionalObjectsController optionalObjectsController;
+
+  @Mock
+  private LockFactory lockFactory;
+
+  @Mock(extraInterfaces = {Initialisable.class, Disposable.class, Startable.class, Stoppable.class})
+  private Processor targetProcessor;
+
+  @SuppressWarnings("deprecation")
+  private MuleContextWithRegistry muleContext;
+
+  private Set<ExtensionModel> extensions;
+
+  private AtomicInteger initializations;
+
+  @Before
+  public void setup() throws Exception {
+    initializations = new AtomicInteger(0);
+
+    declarer = forExtension(EXTENSION_NAME);
+    muleContext = mockContextWithServices();
+    extensions = ImmutableSet.<ExtensionModel>builder()
+        .add(getExtensionModel())
+        .add(mockExtension)
+        .build();
+
+    MessageProcessorChainBuilder messageProcessorChainBuilder = new DefaultMessageProcessorChainBuilder().chain(targetProcessor);
+    DefaultListableBeanFactory beanFactory = new ObjectProviderAwareBeanFactory(null);
+    MuleRegistry mockedRegistry = muleContext.getRegistry();
+
+    when(extensionManager.getExtensions()).thenReturn(extensions);
+    when(muleContext.getExecutionClassLoader()).thenReturn(currentThread().getContextClassLoader());
+    when(muleContext.getExtensionManager()).thenReturn(extensionManager);
+    when(mockedRegistry.lookupObject(MY_FLOW)).thenReturn(messageProcessorChainBuilder);
+    when(mockedRegistry.get(OBJECT_REGISTRY)).thenReturn(new DefaultRegistry(muleContext));
+
+    lazyMuleArtifactContext = createLazyMuleArtifactContextStub(beanFactory);
+
+    doAnswer(a -> {
+      initializations.incrementAndGet();
+      return null;
+    }).when((Initialisable) targetProcessor).initialise();
+  }
+
+
+  @Test
+  @Issue("MULE-18316")
+  public void shouldNotCreateBeansForSameLocationRequest() {
+    Location location = builderFromStringRepresentation(MY_FLOW).build();
+
+    lazyMuleArtifactContext.initializeComponent(location);
+    lazyMuleArtifactContext.initializeComponent(location);
+
+    assertThat(initializations.get(), is(1));
+
+  }
+
+  @Test
+  @Issue("MULE-18316")
+  public void shouldCreateBeansForSameLocationRequestIfDifferentPhaseApplied() throws InitialisationException {
+    Location location = builderFromStringRepresentation(MY_FLOW).build();
+
+    lazyMuleArtifactContext.initializeComponent(location, false);
+    lazyMuleArtifactContext.initializeComponent(location);
+
+    assertThat(initializations.get(), is(2));
+  }
+
+  @Test
+  @Issue("MULE-18316")
+  public void shouldNotCreateBeansForSameLocationFilterRequest() {
+    ComponentLocationFilter filter = loc -> loc.getLocation().equals(MY_FLOW);
+
+    lazyMuleArtifactContext.initializeComponents(filter);
+    lazyMuleArtifactContext.initializeComponents(filter);
+
+    assertThat(initializations.get(), is(1));
+  }
+
+  @Test
+  @Issue("MULE-18316")
```

Comment:
since this is just a move/refactor of tests, the issue should point to the original one, not the refactor one